### PR TITLE
THU/scenario_sharing_prerequisites

### PR DESCRIPTION
**Pull request by csm-thu**
- test: move jest tests fixtures to a separate folder
- feat: add new utils functions getPermissionsFromRoles
- test: rename waitNewPageQueries to waitAliases, add options as parameter
- test: add selectors & actions for roles edition components

### CODE REVIEW

File: src/utils/ScenariosUtils.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+  }
+  if (scenario.ownerId === userId) {
+    return permissionsMapping[ACL_ROLES.SCENARIO.ADMIN];
+  }
+
+  return SecurityUtils.getUserPermissionsForResource(scenario.security, userEmail, permissionsMapping);
+};
+
+const patchScenarioWithCurrentUserPermissions = (scenario, userEmail, userId, permissionsMapping) => {
+  scenario.security = {
```
Comment (ardvi) [inline src/utils/ScenariosUtils.js:20]: Pourquoi ne pas affecter directement scenario.security.currentUserPermissions =  _getUserPermissionsForScenario(scenario, userEmail, userId, permissionsMapping)

**Review by esasova — APPROVED** [review]
LGTM :+1:

File: src/utils/SecurityUtils.js [hunk 2]
```diff
@@ -0,0 +1,206 @@
...
+    const userAccessB = _findById(aclB, userAccessA.id);
+    if (userAccessB === undefined || userAccessA.role !== userAccessB.role) return false;
+  }
+  return true;
+};
+
+const compareAccessControlLists = (currentACL, newACL) => {
+  const entriesToAdd = [];
+  const entriesToRemove = [];
+  for (const newUserAccess of newACL) {
```
Comment (ardvi) [inline src/utils/SecurityUtils.js:20]: for be more readable, maybe we can create a function like getMissingUsersAccess and getUsersRoleChanged.
the result will be :
const newUsersAccess = getMissingUsersAccess(currentAcl, newAcl);
const deletedUsersAccess = getMissingUsersAccess(newAcl, currentAcl);
const userAcccessRoleChanged = getUsersAccessRoleChanged(currentAcl, newAcl);

const entriesToAdd = newUsersAccess.concat(userAcccessRoleChanged);
const entriesToRemove = deletedUsersAccess.concat(userAccessRoleChanged);

File: src/utils/SecurityUtils.js [hunk 3]
```diff
@@ -0,0 +1,206 @@
...
+    }
+  }
+
+  for (const currentUserAccess of currentACL) {
+    const newUserAccess = _findById(newACL, currentUserAccess.id);
+    // Entry removed from ACL
+    if (newUserAccess === undefined) entriesToRemove.push(currentUserAccess);
+  }
+
+  const usersToRemove = entriesToRemove.map((entry) => entry.id);
```
Comment (ardvi) [inline src/utils/SecurityUtils.js:37]: Can you reuse the function getUsersFromAcl here?

File: src/utils/SecurityUtils.js [hunk 4]
```diff
@@ -0,0 +1,206 @@
...
+  return { entriesToAdd, entriesToRemove, usersToRemove };
+};
+
+const getUsersFromACL = (acl) => {
```
Comment (ardvi) [inline src/utils/SecurityUtils.js:41]: in fact is getUsersIdFromAcl no ?

File: src/utils/SecurityUtils.js (hunk 3 above)
Comment (ardvi) [inline src/utils/SecurityUtils.js:37]: usersToRemove > userIdsToRemove i think

Comment (csm-thu) [reply src/utils/SecurityUtils.js:37]: Good point :smile:

File: src/utils/SecurityUtils.js (hunk 2 above)
Comment (csm-thu) [reply src/utils/SecurityUtils.js]: Lisibility has been improved, reduced returned objects to only two :slightly_smiling_face:

**Pull request closed without merging**